### ISSUE TITLE
🐛 BUG: client:load causes hydration sync errors for vue component when using a store

### ISSUE THREAD

**Issue by songololo** [withastro/astro#2658]
### What version of `astro` are you using?

0.23.0

### What package manager are you using?

yarn

### What operating system are you using?

Mac

### Describe the Bug

I have a `vue` component which works as expected when used without a `client:` directive, e.g.
```js
<TestC />
```

But when adding the directive, i.e.
```js
<TestC client:load/>
```

Then it will load OK on the first load, but afterwards starts returning one of the following two types of errors:
```bash
[Vue warn]: Hydration children mismatch in <div>: server rendered element contains more child nodes than client vdom. 
  at <NavTree currentPath="/" > 
  at <App>
Hydration completed but contains mismatches.
```
or
```bash
Uncaught (in promise) Error: Unable to find the root for the component TestC
```

The issue seems to arise because I'm using a nanostore to share information about my blog posts between different components, which means that the server-side state and client-side state can diverge.

Please see the reproducible repo link.

Thanks!

### Link to Minimal Reproducible Example

https://github.com/songololo/astro-debug

**Comment by Bryanoxx** [comment]
Hello ! Juste for reporting: I have the same error/behavior using the v0.23.1

**Comment by kavaro** [comment]
FYI: Same issue with svelte component that uses a svelte store.

**Comment by hayley** [comment]
Also having this issue:
* upgrade from Astro 0.20.12 to 0.25.4
* svelte component
* uses svelte stores
* `client:visible`

**Comment by hayley** [comment]
Astro 0.26 update: For the same project of mine previously mentioned, upgrading to Astro 0.26 has not fixed the issue.

I'm still seeing `Error: Unable to find the root for the component Radar` in the browser console.

**Comment by hayley** [comment]
Astro 0.26.1 fixes this for me both in the Svelte store and nanostores version.

Based on the changelog (https://github.com/withastro/astro/blob/main/packages/astro/CHANGELOG.md), I'm guessing it was #2978 that fixed this.

To anyone affected by this, I'd encourage you to try 0.26.1 / 1.0 beta.0 to see if it fixes it for you.

**Comment by kavaro** [comment]
I can confirm that Astro 0.26.1 fixes my issues with svelte
Thanks guys

**Comment by FredKSchott** [comment]
Sounds like this was fixed in Astro v0.26.1, thanks everyone!
